addons/survey/controllers/main.py

```diff
@@ -728,39 +723,137 @@ def _get_user_input_domain(self, survey, line_filter_domain, **post):
+                filters_domains += self._prepare_matching_answer_domain(input_line_id.answer_type, input_line_id, operator=operator, value=value)
+                search_filters.append(self._prepare_search_filter('L', input_line_id, value=value))
+
+        user_input_domain = self._get_results_page_user_input_domain(survey, **post)
+        user_input_lines = self._get_matching_user_input_lines(filters_domains, user_input_domain)
 
         return user_input_lines, search_filters
+
+    def _get_matching_user_input_lines(self, filters_domains, user_input_domain):
+        """
+            Retrieve the matching user input lines depending on the survey base domain and the filters domains
+            :rtype: recordset
+        """
+        # Add filters domain to the base domain
+        if filters_domains:
+            all_required_lines_domains = [
+                [('user_input_line_ids', 'in', request.env['survey.user_input.line'].sudo()._search(filters_domain))]
+                for filters_domain in filters_domains
+            ]
+            user_input_domain = expression.AND([user_input_domain, *all_required_lines_domains])
+
+        # Get the matching user input lines
+        user_inputs_query = request.env['survey.user_input'].sudo()._search(user_input_domain)
+        user_input_lines = request.env['survey.user_input.line'].search([('user_input_id', 'in', user_inputs_query)])
+
+        return user_input_lines
+
+    @staticmethod
+    def _get_filters_from_post(post):
+        """
+            Extract the filters from post depending on the model that needs to be called to retrieve the filtered answer data.
+            Model = 'A' : Answer data in the survey.question.answer model (simple_choice, multiple_choice, matrix)
+            Model = 'L' : Answer data in the survey.user_input.line model (char_box, text_box, numerical_box, date, datetime)
+
+            :rtype: (defaultdict[List[int]], List[int])
+        """
+        answer_cols_to_rows_mapping = defaultdict(list)
+        answer_user_input_line_ids = []
+
+        for data in post.get('filters', '').split('|'):
+            if not data:
+                break
+            model, row_id, ans_id = data.split(',')
+            row_id, ans_id = int(row_id), int(ans_id)
+            if model == 'A':
+                answer_cols_to_rows_mapping[ans_id].append(row_id)
+            elif model == 'L' and not row_id:
+                answer_user_input_line_ids.append(ans_id)
+
+        return answer_cols_to_rows_mapping, answer_user_input_line_ids
+
+    @staticmethod
+    def _get_value_with_operator(record, row_id=0):
+        """
+            Get the answer's value and domain operator to find answers with the
+            same value (depends on answer's model and question type).
+
+            :param record: 'survey.question.answer' or 'survey.user_input.line' record
+            :rtype: (string, string)
+        """
```

Comment:
Is this less readable?

```suggestion
        """ Get the answer's value and domain operator to find answers with the
        same value (depends on answer's model and question type).

        :param record: 'survey.question.answer' or 'survey.user_input.line' record
        :rtype: (string, string)
        """
```